library/core/functions.general.php

```diff
@@ -3846,3 +3863,22 @@ function urlMatch($pattern, $url) {
         return true;
     }
 }
+
+if (!function_exists('walkAll')) {
+    /**
+     * Recursively walk through all array elements or object properties.
+     *
+     * @param array|object $input
+     * @param callable $callback
+     * @param string $parent Array key or object property to indicate parent node.
+     */
+    function walkAll(&$input, $callback, $parent = null) {
```

Comment:
- The standard PHP practice puts the word "recursive" at the end of these such function names.
- I'm not a huge fan of making the $parent parameter something that the original caller can pass. Perhaps an inner closure that exposes it is better.
- Cycles can create infinite recursion. Sad.